Epsilon/Services/CompetenceDocumentService.cs

```diff
@@ -25,13 +25,14 @@ ILearningOutcomeCanvasResultService canvasResultService
     public async Task<CompetenceDocument> GetDocument(string userId, DateTime? from = null, DateTime? to = null)
     {
         var submissions = _canvasResultService.GetSubmissions(userId);
+        submissions = submissions.Where(static e => e.Criteria.Any());
         if (from != null && to != null)
         {
             submissions = submissions.Where(s => s.SubmittedAt >= from && s.SubmittedAt <= to);
         }
 
         var domains = await _domainService.GetDomainsFromTenant();
-        var components = FetchComponents(submissions, domains);
+        var components = FetchComponents(submissions, domains, await _domainService.GetOutcomes());
```

Comment:
Argument of type 'IEnumerable<LearningDomainOutcome?>' cannot be used for parameter 'outcomes' of type 'IEnumerable<LearningDomainOutcome>' in 'IAsyncEnumerable<AbstractCompetenceComponent> CompetenceDocumentService.FetchComponents(IAsyncEnumerable<LearningDomainSubmission> submissions, IEnumerable<LearningDomain?> domains, IEnumerable<LearningDomainOutcome> outcomes)' due to differences in the nullability of reference types.